### PR TITLE
feat(hatchery): conf as json for /admin/services ui

### PR DESCRIPTION
**Pull request by yesnault**
Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

File: engine/hatchery/openstack/types.go [hunk 1]
```diff
@@ -15,40 +15,40 @@ import (
 
 // HatcheryConfiguration is the configuration for hatchery
 type HatcheryConfiguration struct {
-	hatchery.CommonConfiguration `mapstructure:"commonConfiguration" toml:"commonConfiguration"`
+	hatchery.CommonConfiguration `mapstructure:"commonConfiguration" toml:"commonConfiguration" json:"commonConfiguration"`
 
 	// Tenant openstack-tenant
-	Tenant string `mapstructure:"tenant" toml:"tenant" default:"" commented:"false" comment:"Openstack Tenant, generally value of $OS_TENANT_NAME, v2 auth only"`
+	Tenant string `mapstructure:"tenant" toml:"tenant" default:"" commented:"false" comment:"Openstack Tenant, generally value of $OS_TENANT_NAME, v2 auth only" json:"-"`
```
Comment (fsamin) [inline engine/hatchery/openstack/types.go]: why not ?
and add omitEmpty

File: engine/hatchery/openstack/types.go [hunk 2]
```diff
@@ -15,40 +15,40 @@ import (
...
 
 	// Domain openstack-domain
-	Domain string `mapstructure:"domain" toml:"domain" default:"" commented:"true" comment:"Openstack Domain, generally value of $OS_DOMAIN_NAME, v3 auth only"`
+	Domain string `mapstructure:"domain" toml:"domain" default:"" commented:"true" comment:"Openstack Domain, generally value of $OS_DOMAIN_NAME, v3 auth only" json:"domain"`
```
Comment (fsamin) [inline engine/hatchery/openstack/types.go]: omitEmpty

File: engine/hatchery/openstack/types.go [hunk 3]
```diff
@@ -15,40 +15,40 @@ import (
...
-	Password string `mapstructure:"password" toml:"password" default:"" commented:"false" comment:"Openstack Password"`
+	Password string `mapstructure:"password" toml:"password" default:"" commented:"false" comment:"Openstack Password" json:"-"`
 
 	// Region openstack-region
-	Region string `mapstructure:"region" toml:"region" default:"" commented:"false" comment:"Openstack Region"`
+	Region string `mapstructure:"region" toml:"region" default:"" commented:"false" comment:"Openstack Region" json:"region"`
 
 	// NetworkString openstack-network
-	NetworkString string `mapstructure:"networkString" toml:"networkString" default:"Ext-Net" commented:"false" comment:"Hatchery will use this Network to spawn CDS Worker (Virtual Machine)."`
+	NetworkString string `mapstructure:"networkString" toml:"networkString" default:"Ext-Net" commented:"false" comment:"Hatchery will use this Network to spawn CDS Worker (Virtual Machine)." json:"networkString"`
```
Comment (fsamin) [inline engine/hatchery/openstack/types.go]: omitEmpty

File: engine/hatchery/openstack/types.go [hunk 4]
```diff
@@ -15,40 +15,40 @@ import (
...
 
 	// IPRange IP Range
-	IPRange string `mapstructure:"iprange" toml:"iprange" default:"" commented:"false" comment:"Facultative. IP Range for spawned workers. \n Format: a.a.a.a/b,c.c.c.c/e \n Hatchery will use an IP from this range to create Virtual Machine (Fixed IP Attribute).\nIf not set, it will get an address from the neutron service"`
+	IPRange string `mapstructure:"iprange" toml:"iprange" default:"" commented:"false" comment:"Facultative. IP Range for spawned workers. \n Format: a.a.a.a/b,c.c.c.c/e \n Hatchery will use an IP from this range to create Virtual Machine (Fixed IP Attribute).\nIf not set, it will get an address from the neutron service" json:"iprange"`
```
Comment (fsamin) [inline engine/hatchery/openstack/types.go]: omitEmpty

File: engine/hatchery/swarm/types.go [hunk 5]
```diff
@@ -13,21 +13,21 @@ type HatcheryConfiguration struct {
...
 	// WorkerTTL Worker TTL (minutes)
-	WorkerTTL int `mapstructure:"workerTTL" toml:"workerTTL" default:"10" commented:"false" comment:"Worker TTL (minutes)"`
+	WorkerTTL int `mapstructure:"workerTTL" toml:"workerTTL" default:"10" commented:"false" comment:"Worker TTL (minutes)" json:"workerTTL"`
 
 	// DockerOpts Docker options
-	DockerOpts string `mapstructure:"dockerOpts" toml:"dockerOpts" default:"" commented:"true" comment:"Docker Options. --add-host and --privileged supported. Example: dockerOpts=\"--add-host=myhost:x.x.x.x,myhost2:y.y.y.y --privileged\""`
+	DockerOpts string `mapstructure:"dockerOpts" toml:"dockerOpts" default:"" commented:"true" comment:"Docker Options. --add-host and --privileged supported. Example: dockerOpts=\"--add-host=myhost:x.x.x.x,myhost2:y.y.y.y --privileged\"" json:"dockerOpts"`
 
-	DockerEngines map[string]DockerEngineConfiguration `mapstructure:"dockerEngines" toml:"dockerEngines" comment:"List of Docker Engines"`
+	DockerEngines map[string]DockerEngineConfiguration `mapstructure:"dockerEngines" toml:"dockerEngines" comment:"List of Docker Engines" json:"dockerEngines"`
```
Comment (fsamin) [inline engine/hatchery/swarm/types.go]: omitEmpty

File: engine/hatchery/swarm/types.go [hunk 6]
```diff
@@ -13,21 +13,21 @@ type HatcheryConfiguration struct {
...
-	DefaultMemory int `mapstructure:"defaultMemory" toml:"defaultMemory" default:"1024" commented:"false" comment:"Worker default memory in Mo"`
+	DefaultMemory int `mapstructure:"defaultMemory" toml:"defaultMemory" default:"1024" commented:"false" comment:"Worker default memory in Mo" json:"defaultMemory"`
 
 	// WorkerTTL Worker TTL (minutes)
-	WorkerTTL int `mapstructure:"workerTTL" toml:"workerTTL" default:"10" commented:"false" comment:"Worker TTL (minutes)"`
+	WorkerTTL int `mapstructure:"workerTTL" toml:"workerTTL" default:"10" commented:"false" comment:"Worker TTL (minutes)" json:"workerTTL"`
 
 	// DockerOpts Docker options
-	DockerOpts string `mapstructure:"dockerOpts" toml:"dockerOpts" default:"" commented:"true" comment:"Docker Options. --add-host and --privileged supported. Example: dockerOpts=\"--add-host=myhost:x.x.x.x,myhost2:y.y.y.y --privileged\""`
+	DockerOpts string `mapstructure:"dockerOpts" toml:"dockerOpts" default:"" commented:"true" comment:"Docker Options. --add-host and --privileged supported. Example: dockerOpts=\"--add-host=myhost:x.x.x.x,myhost2:y.y.y.y --privileged\"" json:"dockerOpts"`
```
Comment (fsamin) [inline engine/hatchery/swarm/types.go]: omitEmpty

File: engine/hatchery/swarm/types.go [hunk 7]
```diff
@@ -46,12 +46,12 @@ type dockerClient struct {
...
-	InsecureSkipTLSVerify bool   `mapstructure:"insecureSkipTLSVerify" toml:"insecureSkipTLSVerify" comment:"DOCKER_INSECURE_SKIP_TLS_VERIFY"` // !DOCKER_TLS_VERIFY
-	TLSCAPEM              string `mapstructure:"TLSCAPEM" toml:"TLSCAPEM" comment:"content of your ca.pem"`
-	TLSCERTPEM            string `mapstructure:"TLSCERTPEM" toml:"TLSCERTPEM" comment:"content of your cert.pem"`
-	TLSKEYPEM             string `mapstructure:"TLSKEYPEM" toml:"TLSKEYPEM" comment:"content of your key.pem"`
-	APIVersion            string `mapstructure:"APIVersion" toml:"APIVersion" comment:"DOCKER_API_VERSION"` // DOCKER_API_VERSION
-	MaxContainers         int    `mapstructure:"maxContainers" toml:"maxContainers" default:"10" commented:"false" comment:"Max Containers on Host managed by this Hatchery"`
+	Host                  string `mapstructure:"host" toml:"host" comment:"DOCKER_HOST" json:"host"`                                                                        // DOCKER_HOST
+	CertPath              string `mapstructure:"certPath" toml:"certPath" comment:"DOCKER_CERT_PATH" json:"certPath"`                                                       // DOCKER_CERT_PATH
```
Comment (fsamin) [inline engine/hatchery/swarm/types.go]: hide ?

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#6314.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✘  
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Pull request merged**